### PR TITLE
[GH Actions] Ensure component commits modify component versions

### PR DESCRIPTION
**Pull request by andrewjschuang**
Removes duplicate components from output and prints out better logs.

Context: https://pipedream-users.slack.com/archives/C01E5KCTR16/p1668475452950849

Example output:
Screenshot 2022-11-16 at 4 42 22 PM

### CODE REVIEW

**Comment by dannyroosevelt** [comment]
I'm good if/when @jcortes approves

File: .github/actions/git-diff-on-components/src/index.js [hunk 1]
```diff
@@ -328,22 +333,23 @@ async function run() {
...
     const componentsThatNeedToBeModified = await getComponentsThatNeedToBeModified({ filesToBeCheckedByDependency, otherFiles });
     const componentsPendingForGitDiff = getComponentsPendingForGitDiff(componentsThatNeedToBeModified);
-    componentsDiffContents = await checkVersionModification(componentsPendingForGitDiff);
+    componentsDiffContents = await checkVersionModification(componentsPendingForGitDiff, componentsThatDidNotModifyVersion);
+  }
+
+  const totalErrors = componentsThatDidNotModifyVersion.length + componentsDiffContents.length;
+  let counter = 1;
+
+  if (componentsThatDidNotModifyVersion.length || componentsDiffContents.length) {
```
Comment (jcortes) [inline .github/actions/git-diff-on-components/src/index.js:342]: Here you can also do:
```suggestion
  if (totalErrors) {
```

File: .github/actions/git-diff-on-components/src/index.js [hunk 2]
```diff
@@ -328,22 +333,23 @@ async function run() {
...
-    core.setFailed("You need to increment the version on some components. Please see the output above and https://pipedream.com/docs/components/guidelines/#versioning for more information");
     componentsThatDidNotModifyVersion.forEach((filePath) => {
-      console.log(`You didn't modify the version of ${filePath}`);
+    console.error(`${counter++}) You need to change the version of ${filePath}.`);
     });
-  }
 
-  if (componentsDiffContents.length) {
-    core.setFailed("You need to increment the version on some components that have dependencies on other files.");
     componentsDiffContents.forEach(({ dependencyFilePath, componentFilePath }) => {
```
Comment (jcortes) [inline .github/actions/git-diff-on-components/src/index.js:350]:
```suggestion
  componentsDiffContents.forEach(({ dependencyFilePath, componentFilePath }) => {
```

File: .github/actions/git-diff-on-components/src/index.js [hunk 3]
```diff
@@ -328,22 +333,23 @@ async function run() {
...
-      console.log(`Dependency file ${getComponentFilePath(dependencyFilePath)} was modified but the component's version in ${getComponentFilePath(componentFilePath)} was not`);
+    console.error(`${counter++}) You need to change the version of ${getComponentFilePath(componentFilePath)} since dependency file ${getComponentFilePath(dependencyFilePath)} was modified.`);
     });
```
Comment (jcortes) [inline .github/actions/git-diff-on-components/src/index.js:352]:
```suggestion
  });
```

File: .github/actions/git-diff-on-components/src/index.js [hunk 4]
```diff
@@ -328,22 +333,23 @@ async function run() {
...
+  const totalErrors = componentsThatDidNotModifyVersion.length + componentsDiffContents.length;
+  let counter = 1;
+
+  if (componentsThatDidNotModifyVersion.length || componentsDiffContents.length) {
+    core.setFailed(`You need to increment the version of the ${totalErrors} component(s) below. Please see the output above and https://pipedream.com/docs/components/guidelines/#versioning for more information.`);
   }
 
-  if (componentsThatDidNotModifyVersion.length) {
-    core.setFailed("You need to increment the version on some components. Please see the output above and https://pipedream.com/docs/components/guidelines/#versioning for more information");
     componentsThatDidNotModifyVersion.forEach((filePath) => {
```
Comment (jcortes) [inline .github/actions/git-diff-on-components/src/index.js:346]:
```suggestion
  componentsThatDidNotModifyVersion.forEach((filePath) => {
```

File: .github/actions/git-diff-on-components/src/index.js [hunk 5]
```diff
@@ -328,22 +333,23 @@ async function run() {
...
-      console.log(`You didn't modify the version of ${filePath}`);
+    console.error(`${counter++}) You need to change the version of ${filePath}.`);
     });
```
Comment (jcortes) [inline .github/actions/git-diff-on-components/src/index.js:348]:
```suggestion
  });
```

**Review by jcortes — COMMENTED** [review]
Hi @andrewjschuang I've just added a few minor request changes other than that looks great to me!

**Review by jcortes — COMMENTED** [review]
Hi @andrewjschuang LGTM! Ready for QA!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[GH_Actions_Ensure_component_commits_modify_component_versions_2249.pdf](https://github.com/PipedreamHQ/pipedream/files/10033136/GH_Actions_Ensure_component_commits_modify_component_versions_2249.pdf)

**Review by jcortes — APPROVED** [review]
Approved!

**Pull request merged**